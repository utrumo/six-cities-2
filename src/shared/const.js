const MAX_IMAGE_ON_OFFER_PAGE = 6;
const ASSETS_PATCH = `/`;

const OfferTypeToPresentName = {
  apartment: `Apartment`,
  room: `Private Room`,
  house: `House`,
  hotel: `Hotel`
};

export {
  MAX_IMAGE_ON_OFFER_PAGE,
  ASSETS_PATCH,
  OfferTypeToPresentName
};
