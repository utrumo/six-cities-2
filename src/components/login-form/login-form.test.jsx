import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {SortingVariants} from 'shared/const.js';
import {Provider} from 'react-redux';
import {BrowserRouter as Router} from 'react-router-dom';
import NameSpace from 'store/name-spaces.js';
import LoginForm from './login-form.jsx';

const mockStore = configureStore();

it(`Should renders correctlry`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      currentLocation: `Amsterdam`,
      currentOfferId: 2,
      sortOrder: SortingVariants.POPULAR,
      offers: [
        {
          id: 1,
          city: {
            name: `Amsterdam`,
            location: {
              latitude: 52.37454,
              longitude: 4.897976,
              zoom: 13,
            },
          },
          previewImage: `/img/apartment-01.jpg`,
          images: [`/img/apartment-02.jpg`],
          title: `Beautiful & luxurious apartment at great location`,
          description: `Peaceful studio in the most wanted area in town.`,
          location: {
            latitude: 52.364540000000005,
            longitude: 4.9019759999999994,
            zoom: 16,
          },
          isPremium: true,
          isFavorite: false,
          rating: 4.6,
          price: 120,
          type: `apartment`,
          bedrooms: 5,
          maxAdults: 3,
          goods: [`Washer`],
          host: {
            id: 1,
            name: `Angelina`,
            isPro: true,
            avatarUrl: `img/avatar-angelina.jpg`,
          },
        },
      ],
      offersReviews: [],
    },
    [NameSpace.USER]: {
      isAuthorized: false,
      emailValidationError: `Email validation error`,
      profile: {},
    },
  });
  const tree = renderer
    .create(
        <Provider store={store}>
          <Router>
            <LoginForm onEmailChange={jest.fn()} onSubmit={jest.fn()} />
          </Router>
        </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
