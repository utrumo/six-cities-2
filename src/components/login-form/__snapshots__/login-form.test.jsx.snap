// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should renders correctlry 1`] = `
<form
  className="login__form form"
  onSubmit={[Function]}
>
  <div
    className="login__input-wrapper form__input-wrapper"
  >
    <label
      className="visually-hidden"
    >
      E-mail
    </label>
    <input
      className="login__input form__input"
      name="email"
      onChange={[Function]}
      pattern=".+@.+\\\\..{2,}"
      placeholder="Email"
      required={true}
      type="email"
      value=""
    />
  </div>
  <div
    className="login__input-wrapper form__input-wrapper"
  >
    <label
      className="visually-hidden"
    >
      Password
    </label>
    <input
      className="login__input form__input"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
      value=""
    />
  </div>
  <button
    className="login__submit form__submit button"
    style={Object {}}
    type="submit"
  >
    Sign in
  </button>
</form>
`;
